python-main.js

```diff
@@ -1080,37 +1087,58 @@ function web_editor(config) {
             $("#command-flash > .roundlabel").text(config["translate"]["static-strings"]["buttons"]["command-flash"]["label"]);
             $("#command-flash").attr("title", config["translate"]["static-strings"]["buttons"]["command-flash"]["title"]);
 
-            if (serial){
-              doSerial();
+            if (serial) {
+                doSerial();
             }
         })
-        .catch(function(err) {
-            console.log("Error connecting: " + err);
-            $("#flashing-overlay-container").css("display", "flex");
-            $("#flashing-info").addClass('hidden');
-
-            // If micro:bit does not support dapjs
-            if (err.message === "No valid interfaces found."){
-                $("#flashing-overlay-error").html('<div>' + err + '</div><div>'+ config["translate"]["webusb"]["err"]["update-req"] +'</div><a href="#" onclick="flashErrorClose()">'+ config["translate"]["webusb"]["close"] +'</a>');
-                return;
-            } else if (err.message === "Unable to claim interface.") {
-                $("#flashing-overlay-error").html('<div>'+ config["translate"]["webusb"]["err"]["clear-connect"] +'</div><a href="#" onclick="flashErrorClose()">'+ config["translate"]["webusb"]["close"] +'</a>');
-                return;
-            }
+        .catch(webusbErrorHandler);
+    }
 
-            $("#flashing-overlay-error").html('<div>' + err + '</div><div>'+ config["translate"]["webusb"]["err"]["restart-microbit"] +'</div><a href="#" onclick="flashErrorClose()">'+ config["translate"]["webusb"]["close"] +'</a>');
-        });
+    function webusbErrorHandler(err) {
+        // Disconnect
+        doDisconnect();
+
+        // Error handler
+        $("#flashing-overlay-container").css("display", "flex");
+        $("#flashing-info").addClass('hidden');
+
+        // If micro:bit does not support dapjs
+        var errorMessage;
+        if (err.message === "No valid interfaces found."){
+            errorMessage = "update-req";
+        } else if (err.message === "Unable to claim interface.") {
+            errorMessage = "clear-connect";
+        } else {
+            errorMessage = "reconnect-microbit";
+        }
+
+        // Show error message
+        $("#flashing-overlay-error").html(
+                '<div>' + err.message + '</div><div>' + 
```

Comment:
Okay, so in this case is because `err` here is '"Flash error"', so it was possibly a reject argument rather than an exception:
https://github.com/ARMmbed/dapjs/blob/87f20dcddf9dd026703daed000ae77d3c1764df4/src/daplink/index.ts#L121